api/v1alpha1/common_types.go

```diff
@@ -153,6 +157,10 @@ type WorkloadConfig struct {
 // ClusterStatusConfig defines the observed state of a given cluster. The
 // controllers will generate and populate these fields during reconciliation.
 type ClusterStatusConfig struct {
+	ClusterStatus batchv1.JobConditionType `json:"clusterStatus"`
```

Comment:
This might need `omitempty` to handle existing clusters?

Reply:
Yeah, that would be cleaner. I will change.

On the other hand, Nucleus doesn't complain on the extra field, and CRDs are super-bloated anyway.

Reply:
I'm just thinking this would be where something would break if DCO tries to deserialize an existing cluster and the field is considered "required"